src/commands/tracing.rs

```diff
@@ -0,0 +1,321 @@
+            }
+            TracesSampler::ParentBasedAlwaysOn => {
+                trace::Sampler::ParentBased(Box::new(trace::Sampler::AlwaysOn))
+            }
+            TracesSampler::ParentBasedAlwaysOff => {
+                trace::Sampler::ParentBased(Box::new(trace::Sampler::AlwaysOff))
+            }
+            TracesSampler::ParentBasedTraceIdRatio => trace::Sampler::ParentBased(Box::new(
+                trace::Sampler::TraceIdRatioBased(config.traces_sampler_arg),
+            )),
+        };
+        let resource = Resource::new(vec![KeyValue::new("service.name", "influxdb-iox")]);
+        trace::Config::default()
+            .with_sampler(sampler)
+            .with_resource(resource)
+    };
+
+    match config.traces_exporter {
+        TracesExporter::Jaeger => {
+            let agent_endpoint = format!(
+                "{}:{}",
+                config.traces_exporter_jaeger_agent_host.trim(),
+                config.traces_exporter_jaeger_agent_port
+            );
+            opentelemetry::global::set_text_map_propagator(opentelemetry_jaeger::Propagator::new());
+            Some(
+                opentelemetry_jaeger::new_pipeline()
+                    .with_trace_config(trace_config)
+                    .with_agent_endpoint(agent_endpoint)
+                    .install_batch(opentelemetry::runtime::Tokio)
+                    .unwrap(),
+            )
+        }
+
+        TracesExporter::Otlp => {
+            let jaeger_endpoint = format!(
+                "{}:{}",
+                config.traces_exporter_otlp_host.trim(),
+                config.traces_exporter_otlp_port
+            );
+            Some(
+                opentelemetry_otlp::new_pipeline()
+                    .with_trace_config(trace_config)
+                    .with_endpoint(jaeger_endpoint)
+                    .with_protocol(opentelemetry_otlp::Protocol::Grpc)
+                    .with_tonic()
+                    .install_batch(opentelemetry::runtime::Tokio)
+                    .unwrap(),
+            )
+        }
+
+        TracesExporter::None => None,
+    }
+}
+
+/// An RAII guard. On Drop, tracing and OpenTelemetry are flushed and shut down.
+pub struct TracingGuard(tracing::subscriber::DefaultGuard);
+
+impl Drop for TracingGuard {
+    fn drop(&mut self) {
+        opentelemetry::global::shutdown_tracer_provider();
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum LogFormat {
+    Full,
+    Pretty,
+    Json,
+    Logfmt,
+}
+
+impl std::str::FromStr for LogFormat {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_ascii_lowercase().as_str() {
+            "full" => Ok(Self::Full),
+            "pretty" => Ok(Self::Pretty),
+            "json" => Ok(Self::Json),
+            "logfmt" => Ok(Self::Logfmt),
+            _ => Err(format!(
+                "Invalid log format '{}'. Valid options: full, pretty, json",
```

Comment:
Looks like this is out of date since you added "logfmt":

```suggestion
                "Invalid log format '{}'. Valid options: full, pretty, json, logfmt",
```

You could also use [`clap::arg_enum`](https://docs.rs/clap/2.33.3/clap/macro.arg_enum.html) to do some of this boilerplate for you for all of these enums. I used it on [the object store type](https://github.com/influxdata/influxdb_iox/commit/c7ef18337cdfe20d96934590b91eb81a1f4ea4ba#diff-0f0dc759b584bcbecbadd0bcd8c780a33e726fd8f6e491763febbaf92668d7c4R191) and it seems to do what I wanted it to do.

Reply:
I tried this with another enum, and it treated `AlwaysOn` as such, rather than `always_on`. There may be a directive I missed, because it would make sense that this conversion was built in.